AbletonMCP_Remote_Script/handlers/clips.py

```
@@ -0,0 +1,366 @@
+        }
+    except Exception as e:
+        if ctrl:
+            ctrl.log_message("Error creating clip: " + str(e))
+        raise
+
+
+def add_notes_to_clip(song, track_index, clip_index, notes, ctrl=None):
+    """Add MIDI notes to a clip."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)